### ISSUE TITLE
Show community users which will be automatically joined after the ownership change in the separate tab

### ISSUE THREAD

**Issue by mprakhov** [status-im/status-desktop#12577]
### Description

Based on the https://github.com/status-im/status-desktop/issues/11969 ticket, when we have a community ownership change, all users will be kicked and a new owner will wait until they share their revealed addresses and accept them automatically

Right now such users will be shown in the pending state of the community membership changes.
The idea is to show them in a separate tab

![image](https://github.com/status-im/status-desktop/assets/117639195/d8c0f3d6-5cf2-416b-a073-f0a12f2524e2)

**Comment by mprakhov** [comment]
@benjthayer I need a little help from your side. Where this tab must be placed and how we should call it?

**Comment by benjthayer** [comment]
@mprakhov thanks for reaching out. If all members are kicked, would this screen be visible given all admins would also be kicked?

If admins aren't kicked via this process, my initial instinct was to suggest that we keep the users in the `All members` tab and just give them another treatment / label. They technically are still members from an Admin perspective given they will automatically rejoin as soon as the new owner node comes online. As an admin, I would be worried if my membership count went down to 0 during this interim period, so classing them as technically still members during this phase would be reassuring. I would also still want the ability to kick and ban users during this phase if possible so keeping them in the `All members` tab keeps this screen and the associated `Kick` and `Ban` controls consistent. 

WDYT?

cc @John-44

**Comment by mprakhov** [comment]
@benjthayer All members, including admins and token masters, will be kicked (only the new owner will stay). After auto-rejoin they will receive their roles according to the community permission settings (must be the same roles, if the new owner didn't change anything).

How it will look like on a high level:
1. The owner receives an ownership token and sets himself as a control node of the community
2. All members will be kicked as the new owner requires revealed addresses from all community members
3. Clients will detect, that the ownership changed and send their request to join for auto-rejoin
4. Onwer sees that this user was a member of the community previously and auto accepts him, setting a necessary role based on the community token permissions and revealed address

**Comment by jrainville** [comment]
> If admins aren't kicked via this process, my initial instinct was to suggest that we keep the users in the All members tab and just give them another treatment / label. They technically are still members from an Admin perspective given they will automatically rejoin as soon as the new owner node comes online

I think this makes sense!

The only caveat I would say, and correct me if I'm wrong @mprakhov , when a token master or an admin gets auto-accepted back  in the community, they will **not** have access to the same list of waiting users, so for them, it would look like the community is "empty".
Only the new owner will have the list as far as I know. 

It's not a big deal IMO, we can still go for your idea as it looks better and doesn't need a new tab secction.

**Comment by mprakhov** [comment]
Yes, correct, admins and token master will not see these members. But this can be changed
If we go with the `All members` list, then I need to understand how we want to identify users, who are in a waiting state.
Note: The user can be in a waiting state for the control node when: 
1. He's offline, so he didn't share his revealed address
2. Something went wrong 🥲

**Comment by benjthayer** [comment]
Thanks @jrainville @mprakhov :)

Just to look at this from the angle of trying to make things as simple for the user (owner) as possible, I wonder what benefit there is for the new owner being able to differentiate between members who have auto-rejoined and those that have not yet? 

I know that sounds a bit crazy but just thinking that if the process of rejoining is so automated (assuming permissions have not been changed), that means that whether the user has rejoined the community or not is simply a product of them being online or offline after the new control node has come online, is that correct? And if permissions have changed, they would be kicked and not appear in the members list anyway.

I'm just trying to think of the specific advantage to the owner of being able to differentiate between those members that have rejoined and those that haven't yet given it seems to be so automated and simply be a product of their online/offline status. 

If we do want to amend the UI to flag which of the members are yet to re-share their addresses and rejoin we could do something minimal like the below but I think this is only needed if we can identify the core use case for the owner differentiating between the "ghost" members and those that have rejoined. Off the top of my head, I suppose it would be useful to know how many of my community I will actually reach with `@everyone` type messages etc?

<img width="1440" alt="Members" src="https://github.com/status-im/status-desktop/assets/110033914/3ed8d5ad-e5e3-40c2-a911-52e48cb9d182">

**Comment by jrainville** [comment]
> that means that whether the user has rejoined the community or not is simply a product of them being online or offline after the new control node has come online, is that correct? And if permissions have changed, they would be kicked and not appear in the members list anyway.

Yeah pretty much. Unless the permissions changed, the user can auto rejoin by just going back online.

> If we do want to amend the UI to flag which of the members are yet to re-share their addresses and rejoin we could do something minimal like the below but I think this is only needed if we can identify the core use case for the owner differentiating between the "ghost" members and those that have rejoined. Off the top of my head, I suppose it would be useful to know how many of my community I will actually reach with @everyone type messages etc?

I guess so yes. We wanted something because the people are not actually members until they come back online, so even though the owner could see the members in the management UI, in the chat UI, you won't see anyone.

Knowing who didn't join back can also be helpful in case something went wrong (lost message). That way, the owner can tell the user to leave the community and send a new request to join.

**Comment by benjthayer** [comment]
Thanks @jrainville I've worked up a new option where we show a banner (floating or pinned) on the `All members` view which is interactive - on click it displays a dialog of all members that are yet to rejoin via the new community control node.

I think this works well because it avoids us adding another tab, it also avoids us placing these users in the `Pending` tab which is not technically the right place for them and also allows for the segregation of true members and those that are yet to rejoin. The owner also has the option to remove this info from the UI in the event there are a few stragglers that take weeks to rejoin or if they are simply not interested in this metric.

Let me know what you think and we can take a look with John when he returns :)

Floating banner variant:
https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=44434-265983&mode=design&t=J5xdI3aZ5Gmmi8Ew-4

Pinned banner variant:
https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=44434-236532&mode=design&t=J5xdI3aZ5Gmmi8Ew-4

**Comment by jrainville** [comment]
I think I prefer the pinned banner option out of the two, but I think i still prefer the first option you showed with the clock icon.

It's the easiest to do I think and it's all in the same list. You can easily kick someone from there if they take too long to rejoin.

What do others think?

**Comment by mprakhov** [comment]
I prefer pinned too
Have a few questions: 
1. Number of messages - do we have an API for getting this data? Will it be dropped if the user is kicked and rejoined?
2. If the user is offline and didn't share his address, `date joined` will be absent?

**Comment by benjthayer** [comment]
Hey @jrainville @mprakhov I reviewed the options this morning with John and we went with the below route. I moved the position of the icon to the right of the display name and we've opted to use an hourglass icon given the clock icon may be used in future for an admin-applied timeout for breach of community rules etc.

<img width="1440" alt="Members" src="https://github.com/status-im/status-desktop/assets/110033914/55831d98-39e9-4106-89d5-62ce28e896ff">

Figma designs:
https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=44958-270474&mode=design&t=gMJPVat3M0ZkLv21-4

**Comment by mprakhov** [comment]
One concern from my side:
In the current implementation, the new owner will not have an old request to join so the date joined will be different
How can we solve it - send old requests to join from an old owner to the new one and update only state (waiting for an auto-rejoin)

**Comment by jrainville** [comment]
Anyway, for this ticket, I'd say to only implement the hourglass or a placeholder to show that the user is not re-added yet.
The full UI can be done by the UI team later. Anyway, we do not support the number of messages yet either.
